app/furniture/journal/entry_component.rb

```diff
@@ -9,10 +9,21 @@ def initialize(*args, entry:, **kwargs)
     end
 
     def body_html
-      render_markdown(entry.body)
+      postprocess(render_markdown(entry.body))
     end
 
-    def published_at
+    private def postprocess(text)
+      entry.keywords.map do |keyword|
+        search = keyword.canonical_with_aliases.sort.reverse.join("|")
+        text.gsub!(/\#(#{search})/i) do |match|
+          link_to(match, keyword.location)
+        end
```

Comment:
What about extracting this into a separate (named) helper. Anytime I encounter a regex I find I spend too much time figuring out what it's trying to do.

Reply:
I'm not sure what the name would be; or the right seam... If you have more specific suggestions I'd be down to try something.

Reply:
I pulled out a few small methods with more helpful names, which I think may help? LMK if not.